diff -Nru vzctl-3.0.25.1.orig/bin/Makefile.am vzctl-3.0.25.1/bin/Makefile.am
--- vzctl-3.0.25.1.orig/bin/Makefile.am	2010-12-24 12:45:38.000000000 +0100
+++ vzctl-3.0.25.1/bin/Makefile.am	2011-01-03 10:44:51.000000000 +0100
@@ -48,6 +48,27 @@
 		echo " ***************************************************"; \
 		echo; \
 	fi
+install-pld-from-spec:
+
+install-pld:
+	@if [ ! -e "$(DESTDIR)/sbin/ifup-local" -a \
+			! -L "$(DESTDIR)/sbin/ifup-local" ]; then \
+		$(MKDIR_P) "$(DESTDIR)/sbin"; \
+		$(LN_S) "..$(sbindir)/vzifup-post" \
+			"$(DESTDIR)/sbin/ifup-local"; \
+	elif readlink "$(DESTDIR)/sbin/ifup-local" | \
+			fgrep -q vzifup-post; then \
+		true; \
+	else \
+		echo; \
+		echo " ***************************************************"; \
+		echo " WARNING: file $(DESTDIR)/sbin/ifup-local is present!"; \
+		echo " You have to manually edit the above file so that "; \
+		echo " it calls $(sbindir)/vzifup-post"; \
+		echo " ***************************************************"; \
+		echo; \
+	fi
+
 
 
 install-suse:
diff -Nru vzctl-3.0.25.1.orig/etc/dists/Makefile.am vzctl-3.0.25.1/etc/dists/Makefile.am
--- vzctl-3.0.25.1.orig/etc/dists/Makefile.am	2010-12-24 12:45:35.000000000 +0100
+++ vzctl-3.0.25.1/etc/dists/Makefile.am	2011-01-03 10:44:51.000000000 +0100
@@ -89,6 +89,9 @@
 	$(INSTALL_DATA) $(srcdir)/fedora-core.conf \
 		$(DESTDIR)$(distconfdir)/fedora.conf
 
+pld:
+	$(INSTALL_DATA) $(srcdir)/pld.conf $(DESTDIR)$(distconfdir)/pld.conf
+
 gentoo:
 	$(INSTALL_DATA) $(srcdir)/gentoo.conf $(DESTDIR)$(distconfdir)/gentoo.conf
 
@@ -169,6 +172,7 @@
                    debian \
                    fc \
                    fedora \
+                   pld \
                    gentoo \
                    mandrake \
                    owl \
diff -Nru vzctl-3.0.25.1.orig/etc/init.d/initd-functions.in vzctl-3.0.25.1/etc/init.d/initd-functions.in
--- vzctl-3.0.25.1.orig/etc/init.d/initd-functions.in	2010-12-24 12:45:35.000000000 +0100
+++ vzctl-3.0.25.1/etc/init.d/initd-functions.in	2011-01-05 16:17:46.000000000 +0100
@@ -16,9 +16,12 @@
 
 # Source function library.
 if [ -r /etc/init.d/functions ]; then
-	source /etc/init.d/functions
 	if [ -r /etc/redhat-release ] || [ -r /etc/centos-release ]; then
+		source /etc/init.d/functions
 		DISTR=redhat
+	elif [ -r /etc/pld-release ]; then
+		. /etc/init.d/functions
+		DISTR=pld
 	fi
 elif [ -r /etc/rc.status ]; then
 	source /etc/rc.status
@@ -34,6 +37,8 @@
 {
 	if [ "$DISTR" = "redhat" ]; then
 		echo_success
+	elif [ "$DISTR" = "pld" ]; then
+		ok
 	else
 		echo -n "$rc_done"
 	fi
@@ -42,9 +47,10 @@
 
 print_failure()
 {
-	echo -n "$1"
 	if [ "$DISTR" = "redhat" ]; then
 		failure $"$1"
+	elif [ "$DISTR" = "pld" ]; then
+		fail
 	else
 		echo -n "$rc_failed"
 	fi
@@ -56,6 +62,8 @@
 	if [ "$DISTR" = "redhat" ]; then
 		echo -n "$1"
 		warning $"$1"
+	elif [ "$DISTR" = "pld" ]; then
+		nls -n "$1"
 	else
 		echo -n "- Warning: $1 "
 	fi
@@ -67,15 +75,23 @@
 print_result()
 {
 	if [ $? -eq 0 ] ; then
-		print_success
+		if [ "$DISTR" = "pld" ]; then
+			ok
+		else
+			print_success
+		fi
 	else
-		print_failure "$1"
+		if  [ "$DISTR" = "pld" ]; then
+			fail
+		else
+			print_failure "$1"
+		fi
 	fi
 }
 
 __echo()
 {
-	if [ "$DISTR" = "redhat" ]; then
+	if [ "$DISTR" = "redhat" ] || [ "$DISTR" = "pld" ]; then
 		echo -n $"$1"
 	else
 		echo -n "$1"
@@ -85,7 +101,7 @@
 vzdaemon_start()
 {
 	case $DISTR in
-	   redhat)
+	   redhat|pld)
 		daemon $*
 		;;
 	   suse)
@@ -102,7 +118,7 @@
 vzdaemon_stop()
 {
 	case $DISTR in
-	   redhat|suse)
+	   redhat|suse|pld)
 		killproc $*
 		;;
 	   debian)
@@ -119,7 +135,7 @@
 	shift
 
 	case $DISTR in
-	   redhat|suse)
+	   redhat|suse|pld)
 		status $p
 		;;
 	   debian)
diff -Nru vzctl-3.0.25.1.orig/etc/init.d/Makefile.am vzctl-3.0.25.1/etc/init.d/Makefile.am
--- vzctl-3.0.25.1.orig/etc/init.d/Makefile.am	2010-12-24 12:45:34.000000000 +0100
+++ vzctl-3.0.25.1/etc/init.d/Makefile.am	2011-01-03 10:46:14.000000000 +0100
@@ -15,7 +15,7 @@
 #  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-INITSCRIPTS = vz-redhat vz-gentoo vzeventd-redhat vzeventd-gentoo
+INITSCRIPTS = vz-redhat vz-gentoo vz-pld vzeventd-pld vzeventd-redhat vzeventd-gentoo
 vzlib_DATA = initd-functions
 
 EXTRA_DIST = $(INITSCRIPTS:%=%.in) $(vzlib_DATA:%=%.in)
@@ -29,6 +29,12 @@
 	$(INSTALL_SCRIPT) vz-redhat $(DESTDIR)$(initddir)/vz
 	$(INSTALL_SCRIPT) vzeventd-redhat $(DESTDIR)$(initddir)/vzeventd
 
+install-pld: vz-pld vzeventd-pld
+	$(mkinstalldirs) $(DESTDIR)$(initddir)
+	$(INSTALL_SCRIPT) vz-pld $(DESTDIR)$(initddir)/vz
+	$(INSTALL_SCRIPT) vzeventd-pld $(DESTDIR)$(initddir)/vzeventd
+
+
 install-suse: install-redhat
 
 install-debian: install-redhat
diff -Nru vzctl-3.0.25.1.orig/etc/Makefile.am vzctl-3.0.25.1/etc/Makefile.am
--- vzctl-3.0.25.1.orig/etc/Makefile.am	2010-12-24 12:45:34.000000000 +0100
+++ vzctl-3.0.25.1/etc/Makefile.am	2011-01-03 10:44:51.000000000 +0100
@@ -34,6 +34,11 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(namesdir)
 
+install-pld-from-spec: install-pld
+install-pld:
+	$(MAKE) $(AM_MAKEFLAGS) -C init.d $@
+	$(MAKE) $(AM_MAKEFLAGS) -C network-scripts $@
+
 install-redhat-from-spec: install-redhat
 install-redhat:
 	$(MAKE) $(AM_MAKEFLAGS) -C init.d $@
diff -Nru vzctl-3.0.25.1.orig/etc/network-scripts/Makefile.am vzctl-3.0.25.1/etc/network-scripts/Makefile.am
--- vzctl-3.0.25.1.orig/etc/network-scripts/Makefile.am	2010-12-24 12:45:35.000000000 +0100
+++ vzctl-3.0.25.1/etc/network-scripts/Makefile.am	2011-01-03 10:44:51.000000000 +0100
@@ -38,3 +38,11 @@
 	for file in $(NETSDATA); do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(netsdir)/$$file; \
 	done
+install-pld: $(NETSSCRIPTS)
+	$(mkinstalldirs) $(DESTDIR)$(netsdir)
+	for file in $(NETSSCRIPTS); do \
+		$(INSTALL_SCRIPT) $$file $(DESTDIR)$(netsdir)/$$file; \
+	done
+	for file in $(NETSDATA); do \
+		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(netsdir)/$$file; \
+	done
diff -Nru vzctl-3.0.25.1.orig/Makefile.am vzctl-3.0.25.1/Makefile.am
--- vzctl-3.0.25.1.orig/Makefile.am	2010-12-24 12:45:32.000000000 +0100
+++ vzctl-3.0.25.1/Makefile.am	2011-01-03 10:44:51.000000000 +0100
@@ -46,6 +46,7 @@
 	$(mkinstalldirs) $(DESTDIR)$(modulesdir)
 
 DISTRO_TARGETS = \
+	install-pld \
 	install-redhat \
 	install-gentoo \
 	install-suse \
@@ -53,7 +54,8 @@
 
 DISTRO_TARGETS_ALL = \
 	$(DISTRO_TARGETS) \
-	install-redhat-from-spec
+	install-redhat-from-spec \
+	install-pld-from-spec
 
 $(DISTRO_TARGETS_ALL):
 	$(MAKE) $(AM_MAKEFLAGS) -C etc $@
